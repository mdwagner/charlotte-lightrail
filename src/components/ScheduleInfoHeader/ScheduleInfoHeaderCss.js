import styled from 'styled-components/native';

export const WrapView = styled.View`
  backgroundColor: ${props => props.theme.verticalDividerLine};
  flex: 1;
  height: 74px;
  width: ${props => props.width}px;
`;

export const ScheduleSelectorView = styled.View`
  width: 240px;
  alignSelf: center;
  marginTop: 4px;
  marginBottom: 12px;
`;

export const TitleText = styled.Text`
  alignSelf: center;
  fontSize: 16px;
  marginTop: 9px;
  marginBottom: 3px;
  color: ${props => props.theme.primaryTextColor};
`;
