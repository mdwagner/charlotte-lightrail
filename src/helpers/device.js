import { Dimensions } from 'react-native';

// identify iPhone model based on device width
const identifyDevice = (deviceScreenProps) => {
  if (deviceScreenProps.width === 414) {
    return 'iPhone 6+';
  } else if (deviceScreenProps.width === 320) {
    return 'iPhone 5';
  }
  return 'iPhone 6';
};

const deviceScreen = Dimensions.get('window');
const deviceName = identifyDevice(deviceScreen);
let deviceVariableSizes = {};
let defaultCenter = {};
let defaultZoom;
if (deviceName === 'iPhone 6+') {
  deviceVariableSizes = {
    calloutHeight: 300,
    nextTimeFontSize: 40,
    stationLabelFontSize: 36,
    nextTimeWrapPaddingVertical: 4,
    nearestStationText: 'Nearest Station - '
  };
  defaultCenter = {
    latitude: 35.12848262558094,
    longitude: -80.84703007335676
  };
  defaultZoom = 10.35016331854935;
} else if (deviceName === 'iPhone 6') {
  deviceVariableSizes = {
    calloutHeight: 279,
    nextTimeFontSize: 36,
    stationLabelFontSize: 30,
    nextTimeWrapPaddingVertical: 1,
    nearestStationText: 'Nearest Station - '
  };
  defaultCenter = {
    latitude: 35.09018630471958,
    longitude: -80.84307324707783
  };
  defaultZoom = 9.228717656576594;
} else {
  deviceVariableSizes = {
    calloutHeight: 279,
    nextTimeFontSize: 33,
    stationLabelFontSize: 22,
    nextTimeWrapPaddingVertical: 1,
    nearestStationText: 'Nearest - '
  };
  defaultCenter = {
    latitude: 35.12642689061146,
    longitude: -80.8549019571087
  };
  defaultZoom = 10.10400032344327;
}

const calloutBoxHeight = deviceVariableSizes.calloutHeight - 25; // excludes triangle
const blueBoxHeight = calloutBoxHeight * 0.45; // 45% of calloutBoxHeight
const blackBoxHeight = calloutBoxHeight * 0.55; // 55% of calloutBoxHeight

const insideColPadding = deviceScreen.width * 0.13;
const circleWrapperHeight = 54;
const circleWidth = 40;

export const deviceProps = {
  calloutBoxHeight,
  blueBoxHeight,
  blackBoxHeight,
  deviceVariableSizes,
  deviceScreen,
  defaultCenter,
  defaultZoom,
  deviceName,
  insideColPadding,
  circleWrapperHeight,
  circleWidth
};
