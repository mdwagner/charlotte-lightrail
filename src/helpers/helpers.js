import { Linking } from 'react-native';
import { mapboxApiKey } from './config';

export const startNavigation = async (mode, latlng) => {
  const { latitude, longitude } = latlng;

  const googleMapsURL = `comgooglemaps://?daddr=${latitude},${longitude}&directionsmode=${mode}`;
  const appleMapsURL = `http://maps.apple.com?daddr=${latitude},${longitude}&dirflg=${mode}`;

  const supported = await Linking.canOpenURL(googleMapsURL);
  if (supported) {
    Linking
      .openURL(googleMapsURL)
      .catch(err => console.error('Either Google Maps is not installed or some other error occurred: ', err));
  } else {
    Linking
      .openURL(appleMapsURL)
      .catch(err => console.error('An error occurred: ', err));
  }
};

export const mapboxDistanceAPI = {
  async getDistance(origin, destinations, mode) {
    const coordinates = destinations;
    coordinates.unshift(origin);

    if (__DEV__) {
      const endpoint = `directions-matrix/v1/mapbox/${mode}`;
      coordinates_string = coordinates.map(coordinate => {
        return coordinate.toString();
      });
      coordinates_query = coordinates_string.join(';');
      const url = `https://api.mapbox.com/${endpoint}/${coordinates_query}?access_token=${mapboxApiKey}`;
      const res = await fetch(url);
      return res.json();
    } else {
      const endpoint = `distances/v1/mapbox/${mode}`;
      const url = `https://api.mapbox.com/${endpoint}?access_token=${mapboxApiKey}`;
      const res = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          coordinates,
        }),
      });
      return res.json();
    }
  },
};