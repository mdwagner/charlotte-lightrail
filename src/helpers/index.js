export { withHelpers, withStaticData } from './hoc';