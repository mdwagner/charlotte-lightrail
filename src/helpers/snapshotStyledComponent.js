import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { COLORS } from 'assets/styles/constants';
import renderer from 'react-test-renderer';

export const snapshotStyledComponent = (component) => {
  const snapshot = renderer.create(
    <ThemeProvider theme={COLORS}>
      {component}
    </ThemeProvider>
  ).toJSON()
  expect(snapshot).toMatchSnapshot();
};