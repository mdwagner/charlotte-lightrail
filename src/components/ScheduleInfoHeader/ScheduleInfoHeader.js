import React from 'react';
import PropTypes from 'prop-types';
import { SegmentedControlIOS } from 'react-native';
import { WrapView, TitleText, ScheduleSelectorView } from './ScheduleInfoHeaderCss';
import { withHelpers } from '../../helpers';

class ScheduleInfoHeader extends React.Component {

  static propTypes = {
    scheduleIndex: PropTypes.number.isRequired,
    scheduleValueHandler: PropTypes.func.isRequired,
    stationName: PropTypes.string.isRequired,
    helpers: PropTypes.shape({
      getScheduleDay: PropTypes.func.isRequired,
      deviceProps: PropTypes.shape({
        deviceScreen: PropTypes.object.isRequired
      })
    })
  }

  state = {
    scheduleIndex: this.props.helpers.getScheduleDay().index,
    scheduleValue: this.props.helpers.getScheduleDay().day
  };

  render() {
    const { scheduleIndex, scheduleValueHandler, stationName } = this.props;
    const { deviceScreen } = this.props.helpers.deviceProps;
    return (
      <WrapView width={deviceScreen.width}>
        <TitleText allowFontScaling={false}>{stationName}</TitleText>
        <ScheduleSelectorView>
          <SegmentedControlIOS
            values={['Weekday', 'Saturday', 'Sunday']}
            selectedIndex={scheduleIndex}
            onChange={e => this.setState({ scheduleIndex: e.nativeEvent.selectedSegmentIndex })}
            onValueChange={value => scheduleValueHandler(value)}
            tintColor="white"
          />
        </ScheduleSelectorView>
      </WrapView>
    );
  }
}

export default withHelpers(ScheduleInfoHeader);
