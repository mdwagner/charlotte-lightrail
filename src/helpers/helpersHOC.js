import React from 'react';
import * as configHelpers from './config';
import * as deviceHelpers from './device';
import * as helperHelpers from './helpers';
import * as mapSetupHelpers from './mapSetup';
import * as scheduleHelpers from './scheduleCalcs';

export function withHelpers(Component) {
  return class extends React.Component {
    render() {
      const newProps = {
        ...configHelpers,
        ...deviceHelpers,
        ...helperHelpers,
        ...mapSetupHelpers,
        ...scheduleHelpers
      };
      return (
        <Component {...this.props} helpers={newProps} />
      );
    }
  };
}
