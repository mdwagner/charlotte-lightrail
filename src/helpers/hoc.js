import React from 'react';
import * as configHelpers from './config';
import * as deviceHelpers from './device';
import * as helperHelpers from './helpers';
import * as mapSetupHelpers from './mapSetup';
import * as scheduleHelpers from './scheduleCalcs';
import staticData from './staticData.json';

export function withHelpers(Component) {
  return class extends React.Component {
    constructor(props) {
      super(props);
    }
    render() {
      const newProps = {
        ...configHelpers,
        ...deviceHelpers,
        ...helperHelpers,
        ...mapSetupHelpers,
        ...scheduleHelpers
      };
      return (
        <Component {...this.props} helpers={newProps} />
      );
    }
  };
}

export function withStaticData(Component) {
  return class extends React.Component {
    constructor(props) {
      super(props);
    }
    render() {
      const newProps = {
        staticData
      };
      return (
        <Component {...this.props} staticData={newProps} />
      );
    }
  };
}