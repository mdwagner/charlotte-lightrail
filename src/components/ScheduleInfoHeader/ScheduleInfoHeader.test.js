import React from 'react';
import { snapshotStyledComponent } from '../../helpers/snapshotStyledComponent';
import ScheduleInfoHeader from './ScheduleInfoHeader';
jest.mock('SegmentedControlIOS');
import renderer from 'react-test-renderer';

describe('<ScheduleInfoHeader />', () => {
  let props;

  beforeAll(() => {
    props = {
      scheduleIndex: 1,
      scheduleValueHandler: jest.fn(),
      stationName: 'Test Station'
    }
  });

  it('renders correctly', () => {
    snapshotStyledComponent(
      <ScheduleInfoHeader {...props} />
    );
  });
});