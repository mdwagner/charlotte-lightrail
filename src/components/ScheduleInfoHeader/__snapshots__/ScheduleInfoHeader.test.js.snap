exports[`<ScheduleInfoHeader /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#111113",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": 74,
        "width": 375,
      },
      undefined,
    ]
  }
  width={375}>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#ffffff",
          "fontSize": 16,
          "marginBottom": 3,
          "marginTop": 9,
        },
        undefined,
      ]
    }>
    Test Station
  </Text>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "marginBottom": 12,
          "marginTop": 4,
          "width": 240,
        },
        undefined,
      ]
    } />
</View>
`;
